plugin/testing_upsert.go

```diff
@@ -63,3 +63,50 @@ func (s *WriterTestSuite) testUpsert(ctx context.Context) error {
 
 	return nil
 }
+
+func (s *WriterTestSuite) testUpsertAll(ctx context.Context) error {
+	tableName := fmt.Sprintf("cq_upsert_all_%d", time.Now().Unix())
+	table := schema.TestTable(tableName, s.genDatOptions)
+	table.Columns = append(table.Columns, schema.Column{Name: "name", Type: arrow.BinaryTypes.String, PrimaryKey: true})
+	if err := s.plugin.writeOne(ctx, &message.WriteMigrateTable{
+		Table: table,
+	}); err != nil {
+		return fmt.Errorf("failed to create table: %w", err)
+	}
+
+	records := schema.GenTestData(table, schema.GenTestDataOptions{MaxRows: 1})
```

Comment:
I think we should use `MaxRows: 2` here (and update the expectations accordingly) so that we are also testing a row with null values

Reply:
I did `MaxRows=2` for now but I think it should be removed moving forward as it is a "control flag" which doesn't indicate what it does to the full extant. and we should just have an option `WithNulls` and then just call `GenTestData` twice

Reply:
Agreed, that would be better 👍